src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java

```diff
@@ -725,6 +725,9 @@ public class CustomImportOrderCheck extends AbstractCheck {
     /** Contains objects with import attributes. */
     private final List<ImportDetails> importToGroupList = new ArrayList<>();
 
+    /** Specify format of order declaration customizing by user. */
+    private String customImportOrderRules = "";
+
```

Comment:
So fun... Disabled for a bit and we leaked a problem :).

Reply:
So we come to new nuance: there is setter, but no field.

Reply:
@stoyanK7 , please recheck ability to get content from set method.

Reply:
https://github.com/checkstyle/checkstyle/blob/1ef4cd7e10a71dcbc584ad306049ec355d3888de/src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/CustomImportOrderCheck.java#L804

Not what we need but we need decision to reuse it for xdoc content

Reply:
We should not be allowing pitest to dictate class design/common patterns in Checkstyle. My expectation has always been that a user facing property has an associated field in check class (or superclass).

Reply:
Wait, isn't this list made exactly for this case
https://github.com/checkstyle/checkstyle/blob/1ef4cd7e10a71dcbc584ad306049ec355d3888de/src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java#L138-L152

This is where the test failed
https://github.com/checkstyle/checkstyle/blob/1ef4cd7e10a71dcbc584ad306049ec355d3888de/src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java#L1307-L1317

Reply:
> Wait, isn't this list made exactly for this case

It looks that way, I am not sure if extending this list is going to make document generation any easier, though. For this test in this PR, sure, but my comment at https://github.com/checkstyle/checkstyle/pull/13362#discussion_r1253280784 is more philosophical.

Reply:
> deprecated fields 

Can be removed, they are gone.

Reply:
Let's use doc text from method javadoc, we can remove "Setter that" prefix to get text that we need.

Reply:
To avoid big blockages....

Let's do hack and skip this validation for this module and this property. Create separate issue to comeback to it and fix it. Let's merge this PR with hack.

We will be forced to fix it when we start to migrate/generate properties of modules.

Reply:
Fixing CI right now.